### PR TITLE
Ignore relativeness in patterns

### LINKED ISSUE #6: Ignore relativeness in patterns
Patterns starting with `./`, e.g. `./test/myTest.js` should be treated as `test/myTest.js`. Otherwise they won't match change events from Chokidar, which ends up emitting the changed paths as `test/myTest.js`.

### PR DESCRIPTION
**Pull request by sotojuan**
Fixes #6.

WIP as I'm not sure how to test this as it's more of a Chokidar thing right?

### CODE REVIEW

**Comment by sotojuan** [comment]
I guess I gotta add a test for Coveralls to be happy!

**Comment by novemberborn** [comment]
Actually this won't solve the problem. See https://github.com/avajs/ava-files/issues/6#issuecomment-255593968 for more background. Sorry, should have done more research before opening the issue.

**Comment by sotojuan** [comment]
No problem, I'll see about updating my PR to fix the issue.

**Comment by sindresorhus** [comment]
ping :)

**Comment by sotojuan** [comment]
Completely forgot about this. I just put it on my todo list for this week, thanks for reminding me.

File: index.js [hunk 1]
```diff
@@ -40,6 +40,16 @@ function AvaFiles(options) {
 		files = defaultIncludePatterns();
 	}
 
+	files = files.map(function (file) {
```
Comment (novemberborn) [inline index.js]: I'd move this to line 37 and change it to:

```js
var files = (options.files || []).map(
```

Then remove the `!files ||` part of the condition on line 39 to assign the default patterns. Those don't start with `./` so don't need to be mapped. Focusing the mapping on the `files` that are actually provided seems clearer.

**Comment by sotojuan** [comment]
Done! If it looks good I can squash before merge.

**Comment by sotojuan** [comment]
Squashed.

**Comment by novemberborn** [comment]
Looks good, still [needs a test](https://coveralls.io/builds/9135016/source?filename=index.js#L50) though 😉

**Comment by sotojuan** [comment]
Done!

**Comment by novemberborn** [comment]
:shipit:

**Pull request merged**